arbnode/transaction_streamer.go

```diff
@@ -1443,24 +1457,35 @@ func (s *TransactionStreamer) submitEspressoTransactions(ctx context.Context, ig
 			log.Error("failed to get espresso submitted pos", "err", err)
 			return s.config().EspressoTxnsPollingInterval
 		}
-		bytes, _, err := arbos.ParseEspressoMsg(msg.Message)
+		txns, _, err := arbos.ParseEspressoMsg(msg.Message)
 		if err != nil {
 			log.Error("failed to parse espresso message before submitting", "err", err)
 			return s.config().EspressoTxnsPollingInterval
 		}
 
-		espressoTx := espressoTypes.Transaction{
-			Payload:   bytes[0],
-			Namespace: s.config().EspressoNamespace,
+		if s.dataSigner == nil {
+			log.Error("data signer is nil")
+			return s.config().EspressoTxnsPollingInterval
```

Comment:
I think panic directly may be better

Reply:
Agreed. If something is wrong that we can't handle in the code it's best to exit as early as possible with a clear error message explaining the users why we're crashing.